src/main.js

```diff
@@ -0,0 +1,501 @@
+import { fromBlob, fromUrl, globals, Pool } from "geotiff";
+import { DeferredPromise } from "./DeferredPromise";
+import { Converters } from "./Converters";
+import { parsePerkinElmerChannels } from "./formats/perkinElmer.js";
+
+/**
+ * Enable GeoTIFF Tile Source for OpenSeadragon.
+ *
+ * The GeoTIFFTileSource uses the GeoTIFF.js library to serve tiles from local file or a remote URL.
+ * Remote files require HTTP range requests to be enabled on the server.
+ *
+ * @param {OpenSeadragon} OpenSeadragon - The OpenSeadragon class.
+ */
+export const enableGeoTIFFTileSource = (OpenSeadragon) => {
+  /**
+   * @class GeoTIFFTileSource
+   * @memberof OpenSeadragon
+   * @extends OpenSeadragon.TileSource
+   * @param {File|String|Object} input A File object, url string, or object with fields for pre-loaded GeoTIFF and GeoTIFFImages objects
+   * @param {Object} opts Options object. To do: how to document options fields?
+   *                 opts.logLatency: print latency to fetch and process each tile to console.log or the provided function
+   *                 opts.tileWidth: tileWidth to request at each level. Defaults to tileWidth specified by TIFF file or 256 if unspecified by the file
+   *                 opts.tileHeight:tileWidth to request at each level. Defaults to tileWidth specified by TIFF file or 256 if unspecified by the file
+   *
+   * @property {Object} GeoTIFF The GeoTIFF.js representation of the underlying file. Undefined until the file is opened successfully
+   * @property {Array}  GeoTIFFImages Array of GeoTIFFImage objects, each representing one layer. Undefined until the file is opened successfully
+   * @property {Bool}   ready set to true once all promises have resolved
+   * @property {Object} promises
+   * @property {Number} dimensions
+   * @property {Number} aspectRatio
+   * @property {Number} tileOverlap
+   * @property {Number} tileSize
+   * @property {Array}  levels
+   */
+  class GeoTIFFTileSource extends OpenSeadragon.TileSource {
```

Comment:
As noted above, I think it would be good to have this defined in its own file, for developers who want to import it directly (rather than using it through the npm package)

Reply:
For this use case, do you mean where the plugin is available as just a script tag?

Reply:
I'm thinking of two use cases:

Importing the module directly: 

```
<head>
    <script type="module">
        import { GeoTiffTileSource } from "https://my-favorite-cdn/geotifftilesource/src/geotifftilesource.mjs";

        const tileSource = new GeoTiffTileSource(...);
        viewer.open(tileSource);
    </script>
</head>
```

As a plain script:

```
<head>
    <!-- include openseadragon as global object on the window -->
    <script type="text/javascript" src="openseadragon.js">

    <!-- installs the plugin on the global OpenSeadragon object -->
    <script type="text/javascript" src="https://my-favorite-cdn/geotifftilesource/dist/main.js"> 
    
</head>
```

Thoughts?